### PR TITLE
fix: validation - match references by name and detect all import forms (#3460)

### PR DESCRIPTION
**Pull request by eleshar**
# Bugfix Pull Request

Stacked on #3461 (base `fix/reference-detection-test-isolation`).

## Linked issues

Closes #3460
Relates to #3472

## Context

- Severity/Impact: Low for production (no production code calls these libraries yet), Medium for test signal. This was the last failing suite on develop.
- Affected versions/environments: `scripts/validation/lib/*` reference tooling.

## Reproduction

- Steps: 1) `npx jest --config .jest.config.cjs scripts/validation/__tests__/reference-detection.test.js`
- Expected vs Actual: all pass; 5 failed (one was masked as `it.failing`).

## Root Cause

- `FixSuggester` compared the full path (`agents/issue-agent`) against an index of bare names, so similarity scores were far too low and renames were never suggested. When nothing matched, the result had no `alternativeCandidates` key.
- `BrokenRefsFinder` looked up `agents/agent-one` in a bare-name index, so every `agents/…` import was reported broken. `buildAgentIndex`/`buildSkillIndex` also discarded the names passed to them and scanned the real directories. This bug was not in #3460.
- `ReferenceDetector` missed default and named ES imports.
- The end-to-end test replaced `agents/issue-agent` with the bare name, which is a real edit, but expected no change.

## Fix Summary

- New `reference-name.js` helper:
  - `referenceName()` gives the bare name after the last `agents/` or `skills/`.
  - `replaceReferenceName()` swaps that name and keeps the path and extension.
- `FixSuggester`:
  - Matches on the bare name.
  - Returns the same shape whether or not anything matches.
  - Adds a `replacement` field: the reference with only its name swapped (`agents/issue-triage-agent`).
- `BrokenRefsFinder`:
  - Looks up by name.
  - `buildAgentIndex(names)` and `buildSkillIndex(names)` keep the injected index, as do the new constructor options, matching `FixSuggester`.
- `ReferenceDetector`: covers side-effect, default, named and namespace imports, anchored to the start of a line so prose is not matched.
- Tests:
  - The rename scenario is back from `it.failing` to `it`.
  - The end-to-end test uses `replacement`.
  - New cases cover import forms, prose, the injected index, the no-match shape, and path/extension handling.
- Did not lower the 0.6 similarity threshold (see #3460).

## Verification

- [x] Tests added/updated to cover the bug: `reference-detection` 26/26 and `reference-name` 13/13 pass.
- [x] Manual verification steps: full suite with #3496 applied: **268 of 268 suites pass, 0 failed tests**. Without #3496, the only failures are the 6 suites it fixes.
- [x] Negative/edge cases checked: `skills/js/js.js`, prose mentioning import/from, a dynamic `import()` counted once, a real on-disk agent missing from the injected index.
- /code-review:
  - Applied: the extension-match bug in `replaceReferenceName`, and anchoring imports to a line start.
  - Dismissed: `normalizedValue` flagged as dead code; it is still used for `targetType` and `suggestCorrection`.
- /security-review: no findings. Semgrep: 61 rules, 0 findings.

## Risk & Rollback

- Risk level: Low. The tooling has no production callers.
- Rollback plan: revert this PR.

### Fixed

- **Broken Reference Checks** — Fixed path matching, default and named import detection, and injected test indexes, so all reference detection tests pass. (#3460)

## Summary by CodeRabbit

* **Bug Fixes**
  * Reference validation recognizes side-effect, default, named, and namespace imports, as well as dynamic imports, while avoiding false matches in ordinary prose and unrelated strings.
  * Suggested fixes preserve surrounding paths and file extensions when replacing a reference name, and return consistent empty results when no match is found.
  * Reference checks and fix suggestions use supplied agent and skill indexes consistently, including explicitly empty indexes.

### CODE REVIEW

File: scripts/validation/lib/broken-refs-finder.js [hunk 1]
```diff
@@ -97,9 +111,12 @@ export class BrokenRefsFinder {
...
     const normalizedValue = value.replace(/^\.\//, '').replace(/\.\w+$/, '');
 
+    // Indexes hold bare names, so look up the name, not the path (#3460).
+    const name = referenceName(value);
+
     // Check if reference exists
-    const isAgent = agentIndex.has(normalizedValue);
-    const isSkill = skillIndex.has(normalizedValue);
+    const isAgent = agentIndex.has(name);
+    const isSkill = skillIndex.has(name);
```
[the remark this answers is not in the document]
Comment (eleshar) [reply scripts/validation/lib/broken-refs-finder.js]: Fixed in 3238476. `parseReference()` returns the container; an explicit `agents/` reference is checked against the agent index only and `skills/` against the skill index; a bare name may match either. Tests cover all four cases.

File: scripts/validation/lib/reference-name.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * `agents/issue-agent` renamed to `issue-triage-agent` becomes
+ * `agents/issue-triage-agent`.
+ */
+export function replaceReferenceName(value, newName) {
+  const name = referenceName(value);
+  // Search before any extension, so `skills/js/js.js` never matches `.js`.
+  const extension = value.match(/\.\w+$/)?.[0] ?? '';
+  const index = value.slice(0, value.length - extension.length).lastIndexOf(name);
+  if (index === -1) return newName;
+  return value.slice(0, index) + newName + value.slice(index + name.length);
```
[the remark this answers is not in the document]
Comment (eleshar) [reply scripts/validation/lib/reference-name.js]: Fixed in 3238476. `replaceReferenceName()` now replaces the name segment by offset from the same `parseReference()` that `referenceName()` uses: `agents/foo/foo.js` -> `agents/bar/foo.js`.

File: scripts/validation/__tests__/reference-name.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+describe('replaceReferenceName', () => {
+  it.each([
+    ['issue-agent', 'issue-triage-agent', 'issue-triage-agent'],
+    ['agents/issue-agent', 'issue-triage-agent', 'agents/issue-triage-agent'],
+    ['agents/issue-agent/run.sh', 'triage', 'agents/triage/run.sh'],
+    ['agents/issue-agent/', 'triage', 'agents/triage/'],
+    ['agents/issue-agent/run.test.js', 'triage', 'agents/triage/run.test.js'],
+    ['agents/pr-agent/skills/submit-pr', 'open-pr', 'agents/pr-agent/skills/open-pr'],
+    ['skills/js/js.js', 'bar', 'skills/js/bar.js'],
+    ['agents/js/skills/js/js.js', 'bar', 'agents/js/skills/js/bar.js'],
```
[the remark this answers is not in the document]
Comment (eleshar) [reply scripts/validation/__tests__/reference-name.test.js]: Fixed in 3238476. Expectations corrected (`skills/js/js.js` -> `skills/bar/js.js`, `agents/js/skills/js/js.js` -> `agents/js/skills/bar/js.js`), and every case now asserts `referenceName(result) === newName`. The earlier expectations came from the generated test commit 993e391.

**Comment by eleshar** [comment]
@coderabbitai full review

**Comment by eleshar** [comment]
@coderabbitai full review
